consensus/types/src/beacon_state.rs

```diff
@@ -2142,6 +2174,101 @@ impl<E: EthSpec> BeaconState<E> {
             .map_err(Into::into)
     }
 
+    /// Change the withdrawal prefix of the given `validator_index` to the compounding withdrawal validator prefix.
+    pub fn switch_to_compounding_validator(
+        &mut self,
+        validator_index: usize,
+        spec: &ChainSpec,
+    ) -> Result<(), Error> {
+        let validator = self
+            .validators_mut()
+            .get_mut(validator_index)
+            .ok_or(Error::UnknownValidator(validator_index))?;
+        if validator.has_eth1_withdrawal_credential(spec) {
+            validator.withdrawal_credentials.as_fixed_bytes_mut()[0] =
```

Comment:
Should we use `first_mut` and throw an error if it's `None` here instead of the slice access?

Reply:
I thought of doing it that way first but I couldn't figure out what error to throw in that case. I think it should always be safe because H256 wraps a fixed 32 byte array, so we can always index 0 to 31.
We could also do something like
```rust
let mut bytes = [0; 32];
bytes.copy_from_slice(&validator.withdrawal_credentials.as_bytes());
bytes[0] = spec.compounding_withdrawal_prefix_byte;
validator.withdrawal_credentials = Hash256::from_slice(&bytes);
```
if we want it to be more explicit.

Reply:
Yea i think it's actually fine how it is, `has_eth1_withdrawal_credential` checks the first byte exists anyways